app/shared/hooks/CampaignProvider.js

```diff
@@ -0,0 +1,25 @@
+'use client';
+import { createContext, useState } from 'react';
+import gpFetch from 'gpApi/gpFetch';
+import gpApi from 'gpApi';
+
+export const CampaignContext = createContext([{}, () => {}]);
+
+export const CampaignProvider = ({ children, campaign: initCampaign }) => {
+  const [campaign, setCampaign] = useState(initCampaign);
+  const refreshCampaign = async () => {
+    const { campaign: refreshedCampaign } = await gpFetch(
+      gpApi.campaign.findBySlug,
+      {
+        slug: campaign.slug,
```

Comment:
this endpoint is only allowed for admins.
Normal users can only get the campaign associated with their token (no parameters).
Either make this more general or rename the file/function to AdminCampaignProvider